src/messaging/tests/MessagingContext.cpp

```diff
@@ -23,7 +23,7 @@
 namespace chip {
 namespace Test {
 
-CHIP_ERROR MessagingContext::Init(nlTestSuite * suite, TransportMgrBase * transport)
+CHIP_ERROR MessagingContext::Init(nlTestSuite * suite, TransportMgrBase * transport, bool useDummyMessageCounter)
```

Comment:
bool parameters are generally hard to follow if mixed with other parameters.

How about passing the message counter manager and have a getter. This would mean that instead of:

```cpp
context.Init(suite, transport, true);
context.Init(suite, transport, false);
```

we would have

```cpp
context.Init(suite, transport, context.GetDummyMesssageCounterManager());
context.Init(suite, transport, context.GetMessageCounterManager());
```

and the latter seems clearer and the init code will have one less if (less indent).

Reply:
While looking furthern through the code, I saw that even though these both implement the same protocol, this Init call also initializes the message counter. This seems to couple things too much (knowledge that one requires init while the other does not). We should be able to refactor where the init takes place to have a uniform API.

Reply:
done